WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m

```diff
@@ -473,6 +486,37 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForGeneralSettingsIn
             [self.languageTextCell setTextValue:name];
             return self.languageTextCell;
         }
+        case SiteSettingsGeneralTimeZone:
+        {
+            NSString *timeZoneString;
+            if (self.blog.settings.timeZoneString != nil) {
+                timeZoneString = self.blog.settings.timeZoneString;
+            } else {
+                // timeZoneString is nil, so that means we can't fetch settings for this blog
+                // so instead we'll use the blog options to get the timezone
+                timeZoneString = [self.blog getOptionValue:@"timezone"];
+            }
+            if ([timeZoneString isEqualToString:@""]) {
+                NSNumber *gmtOffset;
+                if (self.blog.settings.gmtOffset != nil) {
+                    gmtOffset = self.blog.settings.gmtOffset;
+                } else {
+                    gmtOffset = [self.blog getOptionValue:@"gmt_offset"];
+                }
+                long hoursUTC = gmtOffset.integerValue;
```

Comment:
Should we use `NSInteger` here instead of `long`, particularly as we're calling `integerValue`?